src/d/actor/d_a_obj_eskban.cpp

```diff
@@ -3,94 +3,416 @@
+    c0.b = tevStr.mColorC0.b;
+    c0.a = tevStr.mColorC0.a;
+
+    JPABaseEmitter* pBEmtr =
+        dComIfGp_particle_set(particleID, &this->current.pos, NULL, NULL, 0xff, NULL, -1,
+                              &tevStr.mColorK0, &c0, &cXyz(1, 1, 1));
+    if (!pBEmtr) {
+        return;
+    }
+    pBEmtr->setGlobalRTMatrix(mpModel->getBaseTRMtx());
 }
 
 /* 00000D44-00000EF0       .text daObjEskban_effect_set__Q211daObjEskban5Act_cFv */
 void daObjEskban::Act_c::daObjEskban_effect_set() {
-    /* Nonmatching */
+    eff_m_break(0x82b1, 2);
+    eff_b_break(0x82b2);
+
+    static cXyz offset_vec(0, 250, 0);
+    mDoMtx_copy(mpModel->getBaseTRMtx(), mDoMtx_stack_c::get());
+    mDoMtx_stack_c::multVec(&offset_vec, &mSmokePos);
+    if (!M_smoke) {
+        return;
+    }
+    JPABaseEmitter* pBEmtr = dComIfGp_particle_setToon(0x2027, &mSmokePos, NULL, NULL, 0xc8,
+                                                       M_smoke, -1, NULL, NULL, NULL);
+    if (!pBEmtr) {
+        return;
+    }
+
+    pBEmtr->setRate(30);
+    pBEmtr->setMaxFrame(1);
+    static JGeometry::TVec3<f32> d_scale(3, 3, 3);
+    static JGeometry::TVec3<f32> p_scale(7, 7, 7);
+    pBEmtr->setGlobalDynamicsScale(d_scale);
+    pBEmtr->setGlobalParticleScale(p_scale);
 }
 
 /* 00000EF0-00001400       .text Execute__Q211daObjEskban5Act_cFPPA3_A4_f */
-void daObjEskban::Act_c::Execute(float(**)[3][4]) {
-    /* Nonmatching */
+int daObjEskban::Act_c::Execute(Mtx** pMtx) {
+    dComIfG_Ccsp()->Set(&mCheckCyl);
+    dComIfG_Ccsp()->Set(&mCameraCyl);
+    dComIfG_Ccsp()->Set(&mCheckSph);
+    if (mCheckSph.ChkCoHit()) {
+        cCcD_Obj* hitObj = mCheckSph.GetCoHitObj();
+        if (hitObj) {
+            fopAc_ac_c* hitAct = hitObj->GetAc();
+            if (hitAct && fpcM_GetName(hitAct) == PROC_NPC_MD) {
+                cXyz dist = fopAcM_GetPosition(hitAct) - fopAcM_GetPosition(this);
+                dist.y = 0;
+                if (dist.normalizeRS()) {
+                    dist *= 10;
+                } else {
+                    dist.set(10, 0, 0);
+                }
+                hitAct->current.pos += dist;
+            }
+        }
+    }
+    switch (mActorState) {
+    case ST_WAIT: /* 0x148 */
+        if (!mCameraCyl.ChkCoHit()) {
+            break;
+        }
+        cCcD_Obj* hitObj;
+        if (!(hitObj = mCameraCyl.GetCoHitObj())) {
+            break;
+        }
+        fopAc_ac_c* hitAct = hitObj->GetAc();
+        if (hitAct && fpcM_GetName(hitAct) == PROC_Bomb2) {
+            mActorID = fpcM_GetID(hitAct);
+            fopAcM_orderOtherEvent(this, "Eskban");
+            mActorState = ST_DESTROYED;
+        }
+        break;
+    case ST_DESTROYED: /* 0x1d8 */
+        if (eventInfo.checkCommandDemoAccrpt()) {
+            mActorState = ST_CUTSCENING;
+            break;
+        }
+        fopAcM_orderOtherEvent(this, "Eskban");
+        break;
+    case ST_CUTSCENING: /* 0x214 */
+        if (!fopAcM_SearchByID(mActorID)) {
```

Comment:
```suggestion
        if (fopAcM_SearchByID(mActorID) == NULL) {
```
this returns a pointer to an actor, not a boolean

Reply:
checking for nullness with `!ptr` is pretty normal c++(98); I do it in other places in this file (and many other files do the same). Do you want the explicit `== NULL` check because this is a return value from a function?

tbh I'm having a bit of a hard time figuring out style expectations without written guidelines even when looking at other files, so apologies if a lot of this doesn't look like it should - hopefully it'll be smoother next time

Reply:
ah alright, I just assumed it was a mistake. in asserts the original devs pretty much always check `var != NULL` (which ends up as `var != 0` in the strings because of double defines), but for if statements it shouldn't matter I think